### PR TITLE
Fix wrong String comparison in EnclosedJsonRecordReader

### PR DESCRIPTION
**Pull request by komamitsu**
`parser.getCurrentName()` should be compared using `String#equals`.

### CODE REVIEW

**Comment by randallwhitman** [comment]
By inspection/review, this patch looks correct.

File: json/src/main/java/com/esri/json/hadoop/EnclosedJsonRecordReader.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public boolean next(LongWritable key, Text value) throws IOException {
 			
 			token = parser.nextToken();
 			
-			while (token != null && !(token == JsonToken.START_ARRAY && parser.getCurrentName() == "features")){
+			while (token != null && !(token == JsonToken.START_ARRAY && parser.getCurrentName().equals("features"))) {
```
Comment (climbage) [inline json/src/main/java/com/esri/json/hadoop/EnclosedJsonRecordReader.java]: Good catch - will `getCurrentName` ever be null in this case?

Comment (komamitsu) [inline json/src/main/java/com/esri/json/hadoop/EnclosedJsonRecordReader.java]: Yes. I added null check.

**Comment by climbage** [comment]
@randallwhitman Looks good to me - merge?

**Comment by randallwhitman** [comment]
Passes code review.  Do we have any tests to run before merge?  @climbage @smambrose

**Comment by randallwhitman** [comment]
Thanks @komamitsu !

**Pull request merged**